src/retyped/modulegen.re

```diff
@@ -148,26 +148,24 @@ and function_type_to_bstype ctx f => {
   | Some _ => raise (ModulegenTypeError (not_supported "Type parameters" ctx))
   | None => ()
   };
-  let params =
```

Comment:
The essence of this change is basically to reuse the type-parsing function from the formal args parse and reuse it for the variadic, if any.

Worth noting that I no longer provide a nominal `[("", BsType.Unit)]` for empty formal args. The compiler tests still seem to pass, which I found strange. Any notion of whether this is somehow wrong or just already handled elsewhere and wasn't necessary?